### PR TITLE
Factor out 'crane' argument from rustFlake

### PR DESCRIPTION
**Pull request by szg251**

### CODE REVIEW

File: flake-lang/flake-rust.nix
Comment (bladyjoker) [inline flake-lang/flake-rust.nix:1]: Leave the `crane` argument! Otherwise you will break users.

Let's make this an experiment in backwards compatibility. 

I suggest you leave the `crane` argument and set it as `crane ? "not set"` and then issue a 

```
builtins.trace ''<current file>: You're setting the `crane` argument which is deprecated and will be removed in the next major revision''
```

This warning is issued if the `crane != "not set"` or smtn like that

**Review by bladyjoker — CHANGES_REQUESTED** [review]

File: flake-lang/flake-rust.nix [hunk 1]
```diff
@@ -17,10 +17,15 @@ pkgs:
...
+
   rustWithTools = pkgs.rust-bin.stable.${rustVersion}.default.override {
     extensions = [ "rustfmt" "rust-analyzer" "clippy" "rust-src" ];
   };
-  craneLib = crane.lib.${pkgs.system}.overrideToolchain rustWithTools;
+  craneLib =
+    if crane == null then
+      crane'.lib.${pkgs.system}.overrideToolchain rustWithTools
+    else
+      throw ''rustFlake: You're setting the `crane` argument which is deprecated and will be removed in the next major revision'';
```
Comment (bladyjoker) [inline flake-lang/flake-rust.nix:28]: I don't think it's `throw` you want to only issue a `warning` (not sure if that's builtins.trace or smtn else) or else this will break everybody.

I think it's ok to ignore the `crane` the user did set and use the 'correct' one.

**Review by bladyjoker — CHANGES_REQUESTED** [review]
TLDR: We don't want to `throw` but issue a warning

**Review by bladyjoker — APPROVED** [review]
